prime-router/src/main/kotlin/fhirengine/translation/hl7/utils/FhirPathUtils.kt

```diff
@@ -103,12 +104,14 @@ object FhirPathUtils : Logging {
             }
         } catch (e: Exception) {
             // This is due to a bug in at least the extension() function
-            logger.error(
-                "Unknown error while evaluating FHIR expression $expression for condition. " +
-                    "Setting value of condition to false.",
-                e
-            )
-            false
+            val msg =  when (e) {
+                is FHIRLexerException -> "Syntax error on filter expression"
```

Comment:
At this level, this can be used in many places. Change this to "Syntax error in FHIR Path expression $expression

Reply:
done